pkg/task/restore.go

```diff
@@ -229,7 +230,13 @@ func RunRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 	}
 	// Always run the post-work even on error, so we don't stuck in the import
 	// mode or emptied schedulers
-	defer restorePostWork(ctx, client, mgr, clusterCfg)
+	restorePostWorkOnce := sync.Once{}
+	restorePostWork := func() {
+		restorePostWorkOnce.Do(func() {
+			restorePostWork(ctx, client, mgr, clusterCfg)
+		})
+	}
```

Comment:
```suggestion
	shouldRestorePostWork := true
	restorePostWork := func() {
		if shouldRestorePostWork {
			shouldRestorePostWork = false
			restorePostWork(ctx, client, mgr, clusterCfg)
		}
	}
```

no point using `sync.Once{}` since the whole thing is not shared 🤣

Reply:
Right, thanks!